demo/full/scripts/controllers/ContentList.jsx

```diff
@@ -272,18 +489,57 @@ class ContentList extends React.Component {
           </div>
         </div>
         {
-          hasTextInput ?
+          (hasTextInput || (isLocalContent && isSavingOrUpdating)) ?
             (
               <div className="custom-input-wrapper">
+                {
+                  isSavingOrUpdating ?
+                    (<div className="update-control">
+                      <TextInput
+                        className={"text-input" + (customContentName === "" ? " need-to-fill" : "")}
+                        onChange={onNameInput}
+                        value={customContentName}
+                        placeholder={"Content name"}
+                      />
+                      <div className="update-control-buttons">
+                        {
+                          (isSavingOrUpdating) ?
+                            (<Button
+                              className={"choice-input-button save-button"}
+                              onClick={
+                                () => onClickValid(
+                                  chosenContent ?
+                                    chosenContent.id :
+                                    undefined
+                                )
+                              }
+                              value={isLocalContent ? "Update content" : "Save content"}
+                            />) :
+                            null
+                        }
+                        {
+                          (isSavingOrUpdating) ?
```

Comment:
Why two times in a row?
I don't remember if it became possible to chain multiple elements without a parent in JSX but even if that's not the case it may be cleaner to just add a parent.